client/src/templates/Challenges/codeally/show.tsx

```diff
@@ -113,7 +113,8 @@ interface ShowCodeAllyProps {
 
 class ShowCodeAlly extends Component<ShowCodeAllyProps> {
   static displayName: string;
-  private _container: RefObject<HTMLElement> | undefined;
+  private container: React.RefObject<HTMLElement> | undefined =
+    React.createRef();
```

Comment:
Do all of these need to be `undefined`? It looks like we **should** be setting the refs in the `constructor`:

```js
class MyComponent extends React.Component {
  constructor(props) {
    super(props);
    this.myRef = React.createRef();
  }
```

Ref about refs: https://legacy.reactjs.org/docs/refs-and-the-dom.html

```suggestion
  private container: React.RefObject<HTMLElement> =
    React.createRef();
```

Reply:
Technically `ref` is `Legacy<T> | undefined` 🤔 because typescript doesn't know if the element render, but you are 100% correct, we don't need it. Because `container` should always exist when we assign `createRef()` to hook into the element, so we can drop `undefined` safely.

Reply:
Worked without issues 👍